src/Microsoft.Identity.Web/WebApiExtensions/RequiredScopePolicyProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Identity.Web
+{
+    internal class RequiredScopePolicyProvider : IAuthorizationPolicyProvider
```

Comment:
I don't think you need to replace the policy provider, this is a global singleton and you should be able to expose your extension method without replacing this implementation.

Reply:
can we still ensure that the existing applications work without changing the code for the authorization?
Today with the filters, there is not even a call to `services.AddAuthorization()` in the templates ...

Reply:
Yes, the gist shows how to enable code to enable your attributes using a default policy.